invenio_rdm_records/requests/user_moderation/utils.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 CERN.
+# Copyright (C) 2023 TU Wien.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""RDM user moderation utilities."""
+
+from invenio_db import db
+from invenio_search.api import RecordsSearchV2
+
+from ...proxies import current_rdm_records_service
+
+
+def get_user_records(user_id, from_db=False, status=None):
```

Comment:
shouldn't this be defined as a service method?

Reply:
After looking into this, it looks like it will require multiple changes to the underlying [`RecordService.search()`](https://github.com/inveniosoftware/invenio-records-resources/blob/e297181296eef56bdfb0d1486c3e570fa741a0aa/invenio_records_resources/services/records/service.py#L241), to allow for controlling the `source` field (i.e. what fields are included in the search results).

I would wait for another use-case where we need to do more flexible querying, before introducing a new set of parameters, since the change now would take a considerable effort.